PSReadLine/ReadLine.cs

```diff
@@ -139,7 +142,12 @@ private void ReadKeyThreadProc()
                 if (handleId == 1) // It was the _closingWaitHandle that was signaled.
                     break;
 
+                var localCancellationToken = _singleton._cancelReadCancellationToken;
                 ReadOneOrMoreKeys();
+                if (localCancellationToken.IsCancellationRequested)
```

Comment:
If cancellation is requested, what should happen with buffered keys (keys pressed after cancellation, but before the next call to PSReadLIne)?

And what should happen to a partially typed command line (before the cancellation, but no Enter) - should that be restored. If you look at how `AcceptAndGetNext` is implemented, you can see it's not hard to restore.

Reply:
If cancellation is requested it'll continue to the next loop and any buffered keys will remain in
`_queuedKeys` until the next key is requested.

Originally any partially typed command lines were just being discarded.  Now the
command line is saved and restored when `ReadLine` is called again.  Thanks for pointing out `AcceptAndGetNext`, I didn't know it would be that easy to implement and it feels *much* better now.